src/components/organisms/topbar/project/manualRun/manualRunSettingsDrawer.tsx

```diff
@@ -108,93 +93,106 @@ export const ManualRunSettingsDrawer = () => {
 	const onEntrypointCreate = (value: string) => {
 		const newFunction = { label: value, value };
 		setFileFunctions((prev) => [...prev, newFunction]);
-		setValue("entrypointFunction", newFunction);
 	};
 
-	return (
-		<Drawer className="p-10" name={DrawerName.projectManualRunSettings} variant="dark">
-			<FormProvider {...methods}>
-				<form onSubmit={handleSubmit(onSubmit)}>
-					<div className="flex items-center justify-between gap-3">
-						<Typography className="text-gray-500">{t("configuration")}</Typography>
-						<div className="flex items-center justify-end gap-6">
-							<Button
-								ariaLabel={tButtons("cancel")}
-								className="p-0 font-semibold text-gray-500 hover:text-white"
-								onClick={() => closeDrawer(DrawerName.projectManualRunSettings)}
-							>
-								{tButtons("cancel")}
-							</Button>
-
-							<Button
-								ariaLabel={tButtons("saveAndRun")}
-								className="border-white px-4 py-2 font-semibold text-white hover:bg-black"
-								disabled={!isValid}
-								type="submit"
-								variant="outline"
-							>
-								<IconSvg className="stroke-white" src={!sendingManualRun ? RunIcon : Spinner} />
-
-								{tButtons("saveAndRun")}
-							</Button>
-						</div>
-					</div>
+	const validateForm = useCallback(
+		(hideErrors?: boolean) => {
+			const result = validateManualRun({
+				filePath,
+				entrypointFunction,
+				params: params || "{}",
+			});
 
-					<div className="relative mt-16">
-						<Controller
-							control={control}
-							name="filePath"
-							render={({ field }) => (
-								<Select
-									aria-label={t("placeholders.selectFile")}
-									dataTestid="select-file"
-									disabled={!filesSelectItems.length}
-									isError={!!errors.filePath}
-									label={t("placeholders.file")}
-									noOptionsLabel={t("noFilesAvailable")}
-									{...field}
-									onChange={(selected) => {
-										field.onChange(selected);
-										updateManualRunConfiguration(projectId!, { filePath: selected! });
-									}}
-									options={filesSelectItems}
-									placeholder={t("placeholders.selectFile")}
-								/>
-							)}
-						/>
-
-						<ErrorMessage>{errors.filePath?.message}</ErrorMessage>
-					</div>
+			if (!result.success) {
+				const errors = Object.fromEntries(
+					Object.entries(result.error.flatten().fieldErrors).map(([key, value]) => [key, value?.join(", ")])
+				);
+
+				if (!hideErrors) {
+					setValidationErrors(errors);
+				}
+				setIsValid(false);
+				return false;
+			}
+			setValidationErrors({});
+			setIsValid(true);
```

Comment:
I think can remove `setIsValid` state and check `setValidationErrors` If his empty `const isEmpty = !Object.entries(validationErrors).length;`
```suggestion
			setValidationErrors({});
```

Reply:
We can't. Between lines 106-115 we have a condition that have 2 indicators.
When we filling the form and choosing the file, we don't want to see the error in the function right away.

Reply:
You are right